packages/vite/src/plugins/plugin.ts

```diff
@@ -0,0 +1,294 @@
+import {
+  CreateNodes,
+  CreateNodesContext,
+  TargetConfiguration,
+  joinPathFragments,
+  offsetFromRoot,
+} from '@nx/devkit';
+import { basename, dirname, extname, join } from 'path';
+import { registerTsProject } from '@nx/js/src/internal';
+
+import { getRootTsConfigPath } from '@nx/js';
+
+import { readTargetDefaultsForTarget } from 'nx/src/project-graph/utils/project-configuration-utils';
+import { getNamedInputs } from '@nx/devkit/src/utils/get-named-inputs';
+import { readdirSync } from 'fs';
+import { ExecutorOptions } from '@nx/js/src/utils/schema';
+
+export interface VitePluginOptions {
+  buildTargetName?: string;
+  testTargetName?: string;
+  serveTargetName?: string;
+  previewTargetName?: string;
+}
+
+export const createNodes: CreateNodes<VitePluginOptions> = [
+  '**/vite.config.{js,ts}',
+  (configFilePath, options, context) => {
+    const projectRoot = dirname(configFilePath);
+    // Do not create a project if package.json and project.json isn't there.
+    const siblingFiles = readdirSync(join(context.workspaceRoot, projectRoot));
+    if (
+      !siblingFiles.includes('package.json') &&
+      !siblingFiles.includes('project.json')
+    ) {
+      return {};
+    }
+
+    options = normalizeOptions(options);
+    const projectName = basename(projectRoot);
+
+    return {
+      projects: {
+        [projectName]: {
+          root: projectRoot,
+          projectType: 'library',
+          targets: buildViteTargets(
+            configFilePath,
+            projectRoot,
+            options,
+            context
+          ),
+        },
+      },
+    };
+  },
+];
+
+function buildViteTargets(
+  configFilePath: string,
+  projectRoot: string,
+  options: VitePluginOptions,
+  context: CreateNodesContext
+) {
+  // Is this needed for some reason?
+  getViteConfig(configFilePath, context);
```

Comment:
I think you will need to find the output directory. Right now it is hardcoded as `joinPathFragments('dist', projectRoot)` but the user could change it.